test/functional/feature_evm_miner.py

```diff
@@ -490,20 +489,158 @@ def same_nonce_transferdomain_and_evm_txs(self):
+                    "gasPrice": 25_000_000_000,
+                    "gas": 30_000_000,
+                }
+            )
+            signed = self.nodes[0].w3.eth.account.sign_transaction(tx, self.ethPrivKey)
+            self.nodes[0].w3.eth.send_raw_transaction(signed.rawTransaction)
+
+        for i in range(10):
+            self.nodes[0].transferdomain(
+                [
+                    {
+                        "src": {
+                            "address": self.address,
+                            "amount": "1@DFI",
+                            "domain": 2,
+                        },
+                        "dst": {
+                            "address": self.ethAddress,
+                            "amount": "1@DFI",
+                            "domain": 3,
+                        },
+                        "nonce": start_nonce_erc55 + i,
+                    }
+                ]
+            )
+
+        tx = contract.functions.loop(10_000).build_transaction(
+            {
+                "chainId": self.nodes[0].w3.eth.chain_id,
+                "nonce": start_nonce + 26,
+                "gasPrice": 25_000_000_000,
+                "gas": 30_000_000,
+            }
+        )
+        signed = self.nodes[0].w3.eth.account.sign_transaction(tx, self.ethPrivKey)
+        self.nodes[0].w3.eth.send_raw_transaction(signed.rawTransaction)
+
+        self.nodes[0].generate(1)
+        for i in range(10):
+            self.nodes[0].transferdomain(
+                [
+                    {
+                        "src": {
+                            "address": self.address,
+                            "amount": "1@DFI",
+                            "domain": 2,
+                        },
+                        "dst": {
+                            "address": self.ethAddress,
+                            "amount": "1@DFI",
+                            "domain": 3,
+                        },
+                        "nonce": start_nonce_erc55 + i,
+                    }
+                ]
+            )
+        self.nodes[0].generate(1)
+
+    def mine_transferdomain_txs(self):
+        self.rollback_to(self.start_height)
+        start_nonce_erc55 = self.nodes[0].w3.eth.get_transaction_count(
+            self.address_erc55
+        )
+        for i in range(10):
+            self.nodes[0].transferdomain(
+                [
+                    {
+                        "src": {
+                            "address": self.address,
+                            "amount": "1@DFI",
+                            "domain": 2,
+                        },
+                        "dst": {
+                            "address": self.ethAddress,
+                            "amount": "1@DFI",
+                            "domain": 3,
+                        },
+                        "nonce": start_nonce_erc55 + i,
+                    }
+                ]
+            )
+        self.nodes[0].generate(1)
```

Comment:
to recreate the crash: do not mine a block, send tx with the reused nonce while first batch is still in the mempool

Reply:
This issue should not be related to mempool but on the miner side.

Reply:
> This issue should not be related to mempool but on the miner side.

could be, but on my side I can not reproduce it if the block is already there. only happens when there is no UTXO left in the address and I reuse the nonce.